corehq/toggles.py

```diff
@@ -808,11 +808,13 @@ def _ensure_valid_randomness(randomness):
 
 def _enable_search_index(domain, enabled):
     from corehq.apps.case_search.tasks import reindex_case_search_for_domain
+    from corehq.apps.es import CaseSearchES
     from corehq.pillows.case_search import domains_needing_search_index
     domains_needing_search_index.clear()
-    if enabled:
-        # action is no longer reversible due to roll out of EXPLORE_CASE_DATA
-        # and upcoming migration of CaseES to CaseSearchES
+
+    has_case_search_cases = CaseSearchES().domain(domain).count() > 0
+    if enabled and not has_case_search_cases:
+        # action is not reversible, we want all projects here eventually
```

Comment:
I hadn't seen this change before. Do you think we should remove the `delete_case_search_cases_for_domain` here? (Not for this PR)

https://github.com/dimagi/commcare-hq/blob/8224a32e34d5e8493d9cbc92193334beb68f295c/corehq/apps/case_search/models.py#L177-L192

Reply:
Hmm yeah, that's a good point, probably.  And it looks like the `enable_case_search_for_ecd` command might also need an update like what I added here.

https://github.com/dimagi/commcare-hq/blob/703310fd517e34ffc74ee1b42389f6620f05b960/corehq/apps/reports/management/commands/enable_case_search_for_ecd.py#L53-L55

If the `save_fn` doesn't delete the index when the FF is disabled, and we do the same for `disable_case_search`, then there would be migrated domains that have data in the case search index.

_However_, if no feature flags are enabled, then `domains_needing_search_index` won't include that domain, and I assume the pillow won't process new case changes in those domains.  Honestly it sounds like we need a single place to store whether or not the index is enabled, one which isn't tied to user-facing features, and which is never turned off (`ECD_MIGRATED_DOMAINS`?).  Think I might punt that to https://dimagi-dev.atlassian.net/browse/USH-904

Reply:
👍🏽